homeassistant/components/androidtv_remote/media_player.py

```diff
@@ -0,0 +1,163 @@
+
+    def _update_volume_info(self, volume_info: dict[str, str | bool]) -> None:
+        """Update volume info."""
+        self._attr_volume_level = int(volume_info["level"]) / 100
+        self._attr_is_volume_muted = bool(volume_info["muted"])
+
+    @property
+    def state(self) -> MediaPlayerState:
+        """Return the state of the device."""
+        if self._attr_is_on:
+            return MediaPlayerState.ON
+        return MediaPlayerState.OFF
+
+    async def async_turn_on(self) -> None:
+        """Turn the Android TV on."""
+        if not self._attr_is_on:
+            self._send_key_command("POWER")
+
+    async def async_turn_off(self) -> None:
+        """Turn the Android TV off."""
+        if self._attr_is_on:
+            self._send_key_command("POWER")
+
+    async def async_volume_up(self) -> None:
+        """Turn volume up for media player."""
+        self._send_key_command("VOLUME_UP")
+
+    async def async_volume_down(self) -> None:
+        """Turn volume up for media player."""
+        self._send_key_command("VOLUME_DOWN")
+
+    async def async_mute_volume(self, mute: bool) -> None:
+        """Mute the volume."""
+        if mute != self.is_volume_muted:
+            self._send_key_command("VOLUME_MUTE")
+
+    async def async_set_volume_level(self, volume: float) -> None:
+        """Set volume level, range 0..1."""
+        if isinstance(self._attr_volume_level, float):
+            diff = volume - self._attr_volume_level
+            for _ in range(abs(round(diff * 100))):
+                self._send_key_command("VOLUME_UP" if diff > 0 else "VOLUME_DOWN")
+                await asyncio.sleep(0.1)
+
+    async def async_media_play(self) -> None:
+        """Send play command."""
+        self._send_key_command("MEDIA_PLAY")
+
+    async def async_media_pause(self) -> None:
+        """Send pause command."""
+        self._send_key_command("MEDIA_PAUSE")
+
+    async def async_media_play_pause(self) -> None:
+        """Send play/pause command."""
+        self._send_key_command("MEDIA_PLAY_PAUSE")
+
+    async def async_media_stop(self) -> None:
+        """Send stop command."""
+        self._send_key_command("MEDIA_STOP")
+
+    async def async_media_previous_track(self) -> None:
+        """Send previous track command."""
+        self._send_key_command("MEDIA_PREVIOUS")
+
+    async def async_media_next_track(self) -> None:
+        """Send next track command."""
+        self._send_key_command("MEDIA_NEXT")
+
+    async def async_play_media(
+        self, media_type: MediaType | str, media_id: str, **kwargs: Any
+    ) -> None:
+        """Play a piece of media."""
+        if media_type == MediaType.CHANNEL:
+            if not media_id.isnumeric():
+                raise ValueError(f"Channel must be numeric: {media_id}")
+            for digit in media_id:
+                self._send_key_command(f"KEYCODE_{digit}")
+                await asyncio.sleep(0.1)
```

Comment:
Same as earlier comment. Remove or reduce delay and extract it to a constant in this file used in both places.